x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/rule_preview/preview_table_cell_renderer.tsx

```diff
@@ -5,11 +5,22 @@
  * 2.0.
  */
 
-import type React from 'react';
+import React from 'react';
 import type { EuiDataGridCellValueElementProps } from '@elastic/eui';
+import { TableId } from '@kbn/securitysolution-data-table';
 import type { CellValueElementProps } from '../../../../../common/types';
+import { SourcererScopeName } from '../../../../common/store/sourcerer/model';
 import { RenderCellValue } from '../../../../detections/configurations/security_solution_detections';
 
 export const PreviewRenderCellValue: React.FC<
   EuiDataGridCellValueElementProps & CellValueElementProps
-> = (props) => RenderCellValue({ ...props, asPlainText: true });
```

Comment:
It seems like the underlying issue here was that we were [invoking the component directly](https://kentcdodds.com/blog/dont-call-a-react-function-component), and while that's technically allowed (and was working, without type errors), the recent wrapping of `RenderCellValue` in `memo` changed its return signature in a way that broke the direct invocation. 

Using `createElement` and/or JSX would have prevented these issues, and so invoking the component directly is something we should avoid, since Typescript doesn't appear to offer any assistance here.

I was trying to find an eslint rule that we could add for this purpose, but it seems like [a hard situation to identify](https://github.com/jsx-eslint/eslint-plugin-react/issues/3453), so perhaps we just need to watch out for (and prevent) these patterns as we see them.

cc @marshallmain